speedtest/datadog_checks/speedtest/speedtest.py

```diff
@@ -0,0 +1,163 @@
+import json
+import subprocess
+from typing import Any, Dict
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+
+MEGABYTE_TO_MEBIBYTE = 1048576.0 / 1000000.0
+
+
+class SpeedtestCheck(AgentCheck):
+    def check(self, *args, **kwargs):
+        # type: (Dict[str, Any]) -> None
+        host = self.instance.get("host")
+        ip = self.instance.get("ip")
+        interface = self.instance.get("interface")
+        server_id = self.instance.get("server_id")
+        tags = self.instance.get("tags") or []
+
+        # test config for only **one** of the following above options
+        c = 0
+        for opt in (host, ip, interface, server_id):
+            c += int(opt not in [None, ""])  # trick, int(bool) returns 0 if False, 1 if True
+        if c > 1:
+            raise ConfigurationError("Only one of `host`, `ip`, `interface` or `server_id` may be configured.")
+
+        # Build command
+        cmd = self._build_command(host, ip, interface, server_id)
+
+        # call to speedtest-cli on the system path
+        try:
+            payload = self._call_command(cmd)
+            self._submit_data(payload, tags)
+            self.service_check("speedtest.run", self.OK)
+        except Exception as e:
+            # something went wrong with this command
+            self.log.exception(e)
+            self.service_check("speedtest.run", self.CRITICAL, message=str(e))
+
+    def _build_command(self, host, ip, interface, server_id):
+        # Build command
+        cmd = "speedtest-cli -f json -p no -A -P 8 "
```

Comment:
The tool seems to get installed as just `speedtest` when using homebrew, and when downloading the package directly via https://www.speedtest.net/apps/cli#linux-flyout

Reply:
ack, yup, it seems I have hacked a link on my machine, will update